com.unity.render-pipelines.high-definition/Runtime/Material/MaterialEvaluation.hlsl

```diff
@@ -36,6 +36,105 @@ void AccumulateIndirectLighting(IndirectLighting src, inout AggregateLighting ds
     dst.indirect.specularTransmitted += src.specularTransmitted;
 }
 
+#define ENABLE_PACKED_ACCUMULATION_DATA 1
+
+struct PackedHalf3
+{
+#if ENABLE_PACKED_ACCUMULATION_DATA
+    uint3 m_data;
+
+    float3 first() { return float3(f16tof32(m_data.x), f16tof32(m_data.y), f16tof32(m_data.z)); }
+    float3 second() { return float3(f16tof32(m_data.x >> 16), f16tof32(m_data.y >> 16), f16tof32(m_data.z >> 16)); }
+
+    void Pack(float3 first, float3 second)
+    {
+        m_data = uint3(
+            f32tof16(first.x) | (f32tof16(second.x) << 16),
+            f32tof16(first.y) | (f32tof16(second.y) << 16),
+            f32tof16(first.z) | (f32tof16(second.z) << 16)
+        );
+    }
+#else
+    float3 m_first;
+    float3 m_second;
+    float3 first() { return m_first; }
+    float3 second() { return m_second; }
+    void Pack(float3 first, float3 second)
+    {
+        m_first = first;
+        m_second = second;
+    }
+#endif
+};
+
+struct PackedDirectLighting
+{
+    PackedHalf3 m_data;
+    float3 diffuse () { return m_data.first(); }
+    float3 specular() { return m_data.second(); }
+    void Pack(float3 diffuse, float3 specular)
+    {
+        m_data.Pack(diffuse, specular);
+    }
+};
+
+struct PackedIndirectLighting
+{
+    PackedHalf3 m_data;
+
+    float3 specularReflected() { return m_data.first(); }
+    float3 specularTransmitted() { return m_data.second(); }
+
+    void Pack(float3 specularReflected, float3 specularTransmitted)
+    {
+        m_data.Pack(specularReflected, specularTransmitted);
+    }
+};
+
+struct PackedAggregateLighting
+{
+    float2 m_multipliers;
+    PackedDirectLighting direct;
+    PackedIndirectLighting indirect;
+
+    void init(float exposureMultiplier, float exposureMultiplierInv)
```

Comment:
Note: I am not sure that our various platform support function declaration of function in the struct. We need to validate that

Reply:
Also nitpicking here, should be capitalized (Init)

Reply:
I am confident they do since it should be part of the hlsl spec. However it will take a significant effort to verify unless this is deployed in the CI.
I will just take the conservative approach and do it in a glsl friendly way :). I might also put all this in its own utility header to keep it clean. This could also make it easier in the future to move it to hardware 16 bit.